client/app/app.js

```diff
@@ -202,34 +202,125 @@ angular
+            }
+
+            // build child states that need to be ignored
+            // 1st: all child states of each states that need to be ignored
+            // 2nd: current layout state doesn't need to have itself as modal child
+            layoutState.layout.ignoreChilds.forEach((childState) => {
+                layoutState.layout.ignoreChilds = layoutState.layout.ignoreChilds.concat(_.map(getChildStates(childState), "name"));
+            });
+            layoutState.layout.ignoreChilds.push(layoutState.name);
+
+            // remove child states that need to be ignored
+            childStates = _.xor(childStates, layoutState.layout.ignoreChilds);
+
+            // create child state with layout settings applied
+            const modalSateSuffix = _.last(layoutState.name.split("."));
+            childStates.forEach((childState) => {
+                $stateRegistry.register({
+                    name: `${childState}.${modalSateSuffix}`,
+                    url: layoutState.self.url,
+                    templateUrl: layoutState.self.templateUrl,
+                    controller: layoutState.self.controller,
+                    layout: { // don't know why full config must be defined???
+                        name: "modal",
+                        toChilds: false,
+                        ignoreChilds: []
+                    }
+                });
+            });
+        });
+
+        $urlService.listen(); // Start responding to URL changes
+        $urlService.sync(); // Find the matching URL and invoke the handler
+    })
+    .config(($urlServiceProvider, $stateProvider, $transitionsProvider) => {
         let modalInstance = null;
 
+        /**
+         *  Delay initial URL synchronization.
+         *  As we will create some states later (at run phase), maybe the url we want to access is not yet configured. So wait a moment that all states are declared (at config phase) and re-launch the URL sync later.
+         */
+        $urlServiceProvider.deferIntercept();
+
+        /**
+         *  Create a decorator for our new state attribute 'layout'.
+         *  For modal layout, the attribute can be a string or an object:
+         *  - if string - value must be 'modal'
+         *  - if object - avaiable attributes are:
+         *      - name: the value must be 'modal'.
+         *      - toChilds (default value: 'false'):
+         *          - can be a boolean: 'true' to declare the modal state to all direct parent childs.
+         *          - or an array of string that contains the childs of the direct parents where the modal needs to be displayed.
+         *      - ignoreChilds: an array of string that contains states names where the modal state doesn't need to be displayed.
+         */
+        $stateProvider.decorator("layout", (state) => {
+            let modalLayout;
+
+            if (state.self.layout === "modal" || _.get(state, "self.layout.name") === "modal") {
```

Comment:
If you want to be safe, you should check that `state.self` is defined :)

Reply:
or using `_.get(state, "self.layout")`